server/api/users.js

```diff
@@ -38,20 +38,60 @@ router.get('/:id', async (req, res, next) => {
 router.get('/:id/order-history', async (req, res, next) => {
   try {
     if (
-      (req.session.userId && req.session.userId == req.params.id) ||
+      (req.session.userId && req.session.userId === req.params.id) ||
       req.session.userIsAdmin
     ) {
-      const orders = await Order.findAll({
-        where: {
-          userId: req.params.id,
-          isPurchased: true
-        }
-      })
-      res.json(orders)
+      // const orders = await Order.findAll({
+      //   where: {
+      //     userId: req.params.id,
+      //     isPurchased: true
+      //   }
+      // })
+      const user = await User.findById(req.params.id)
+      const orderHistory = await user.getOrderHistory()
+      res.json(orderHistory)
     } else {
       res.sendStatus(401)
     }
   } catch (err) {
     next(err)
   }
 })
+
+router.get('/:id/shopping-cart', async (req, res, next) => {
+  try {
+    if (
+      (req.session.userId && req.session.userId === req.params.id) ||
```

Comment:
Condition will always fail because of number to string comparison.
Use Number(req.params.id)